### ISSUE TITLE
Page Actions should use constants

### ISSUE THREAD

**Issue by netniV** [Cacti/cacti#1984]
Most listing pages have a dropdown action box.  Currently, this utilises hardcoded number to determine the action to take place.  This should be updated to utilise a constant so that other pages can more easily link to the specified action, and to avoid issues when the order of the list is adjusted.

```php
function form_actions() {
        global $template_actions, $config;

        if (isset_request_var('selected_items')) {
                $selected_items = unserialize(stripslashes(get_request_var('selected_items')));

                if (get_request_var('drp_action') == '1') { // DELETE REPORT TEMPLATE
```

should become something like

```php
function form_actions() {
        global $template_actions, $config;

        if (isset_request_var('selected_items')) {
                $selected_items = unserialize(stripslashes(get_request_var('selected_items')));

                if (get_request_var('drp_action') == REPORTIT_TEMPLATES_DO_DELETE_REPORT) { // DELETE REPORT TEMPLATE
```

Naming convention to be decided upon.

**Comment by netniV** [comment]
I think this could be taken a step further to ensure that any actions taken where actually valid

```php
define('ACTION_POLLERS_ADD',1);
define('ACTION_POLLERS_DELETE',2);
define('ACTION_POLLERS_DUPLICATE',3);
define('ACTION_POLLERS_FULLSYNC',4);

define('ACTION_SITE_ADD',1);
define('ACTION_SITE_DELETE',2);
define('ACTION_SITE_DUPLICATE',3);
```
maybe utilising an array $page_actions in the following structure:

```php
$page_actions = array(
	'pollers.php' => array(
		ACTION_POLLERS_ADD,
		ACTION_POLLERS_DELETE,
		ACTION_POLLERS_DUPLICATE,
		ACTION_POLLERS_FULLSYNC
	),
	'sites.php' => array(
		ACTION_SITES_ADD,
		ACTION_SITES_DELETE,
		ACTION_SITES_DUPLICATE,
	),
);
```

This would then allow us to verify the ACTION value and reset to a default blank value if it was invalid or simply return an error.

**Comment by TheWitness** [comment]
I would try to limit the weight of the global functions.  Things like global_form.php, should be dec-ontented/abandoned and the form's moved into their respective libraries/classes.  Additionally, if the constant is only used in one file, and does not have an api that forces itself to be shared, then it should not be.

We should be cutting out the fat where possible, while keeping the individual libraires readable.

**Comment by netniV** [comment]
OK, so we do that per page rather than as a global but have set_default_action validate it anyway.

**Comment by TheWitness** [comment]
I guess you could make it the second optional parameter to the function.

**Comment by netniV** [comment]
Yeah, or non-optional since any page setting an action must have an action list, even if it's empty?

**Comment by TheWitness** [comment]
Trying to be as generic as possible here.  This is the start.

![image](https://github.com/user-attachments/assets/d2a9bf6d-313a-4b94-b6e0-b11e3d2811d4)

**Comment by netniV** [comment]
I think you missed IMPORT :)

**Comment by TheWitness** [comment]
I'll add it, though it's normally a button.
